library/src/main/java/com/fraggjkee/smsconfirmationview/SmsConfirmationView.kt

```diff
@@ -100,6 +112,55 @@ class SmsConfirmationView @JvmOverloads constructor(
         }
     }
 
+    // Used to inflate and return the popup paste menu
+    private fun initPasteMenu() : PopupMenu {
+        val clipboard = getSystemService(context, ClipboardManager::class.java)
+        val pasteMenu = PopupMenu(this.context, this).apply {
+            inflate(R.menu.popup_menu_paste)
+            setOnMenuItemClickListener { menuItem ->
+                when (menuItem.itemId) {
+                    R.id.menu_item_paste -> {
+                        // Examines the item on the clipboard. If getText() doesn't return null,
+                        // the clip item contains the text. Assumes that this application can only
+                        // handle one item at a time.
+                        val item = clipboard?.primaryClip?.getItemAt(0)
+
+                        // Get the clipboard item text.
+                        val pasteData = item?.text?.toString() ?: ""
+
+                        // Paste the text into the component
+                        appendPaste(pasteData)
+
+                        // Return true/false
+                        pasteData.isBlank()
+                    }
+                    else -> { false }
+                }
+            }
+        }
+
+        // Get the ID of the "paste" menu item.
+        val pasteItem = pasteMenu.menu.findItem(R.id.menu_item_paste)
+
+        // If the clipboard doesn't contain data, disable the paste menu item.
+        // If it does contain data, decide whether you can handle the data.
+        pasteItem.isEnabled = when {
```

Comment:
idea: how about not showing the popup menu at all if there is nothing we can do from there? I think the default OS popup menu just doesn't show "Paste" if the Clibpobar is empty, we can probably follow this here as well